stellargraph/data/explorer.py

```diff
@@ -386,32 +412,29 @@ def run(
         nodes = self.graph.node_ids_to_ilocs(nodes)
 
         if weighted:
-            # Check that all edge weights are greater than or equal to 0.
-            # Also, if the given graph is a MultiGraph, then check that there are no two edges between
-            # the same two nodes with different weights.
-            for node in self.graph.nodes():
-                # TODO Encapsulate edge weights
-                for neighbor in self.graph.neighbors(node, use_ilocs=True):
-
-                    wts = set()
-                    name = f"Edge weight between ({node}) and ({neighbor})"
-                    for weight in self.graph._edge_weights(node, neighbor):
-                        weight_is_valid = (
-                            isinstance(weight, (float, int))
-                            and np.isfinite(weight)
-                            and weight >= 0
-                        )
-                        if not weight_is_valid:
-                            raise ValueError(
-                                f"{name}: expected numeric value greater than or equal to 0, found {weight}"
-                            )
-                        wts.add(weight)
-                    if len(wts) > 1:
-                        # multigraph with different weights on edges between same pair of nodes
-                        raise ValueError(
-                            f"{name}: expected all edges between two particular nodes to have the "
-                            f"same weight value, found {list(wts)}"
-                        )
+            self._check_weights_valid()
+
+            # calculate the appropriate unnormalised transition probability, given the history of
+            # the walk
+            def transition_probability(nn, current_node):
```

Comment:
`current_node` isn't used in this function

Reply:
Ah, left over.